components/index_embeddings/src/main.py

```diff
@@ -0,0 +1,69 @@
+import logging
+
+import dask.dataframe as dd
+import weaviate
+from fondant.component import DaskWriteComponent
+from weaviate.util import _capitalize_first_letter
+
+logger = logging.getLogger(__name__)
+
+
+class IndexEmbeddingsComponent(DaskWriteComponent):
+    def __init__(
+        self,
+        *_,
+        weaviate_url: str,
+        model: str,
+        dataset: str,
+        vectorizer: dict,
+        overwrite: bool,
+    ):
+        self.client = weaviate.Client(weaviate_url)
+        self.overwrite = overwrite
+
+        # following weaviate convention for upper case class names
+        self.class_name = _capitalize_first_letter(f"{model}_{dataset}")
+        self.class_obj = {
+            "class": self.class_name,
+            "properties": [
+                {
+                    "name": "passage",
+                    "dataType": ["text"],
+                },
+                {  # id of the passage in the passage dataset not to mix up with weaviate's uuid
+                    "name": "identifier",
+                    "dataType": ["text"],
+                },
+            ],
+        }
+
+        self.class_obj.update(vectorizer)
+
+    def write(self, dataframe: dd.DataFrame):
+        # using weaviate instead of its langchain wrapper
+        # because langchain does not yet provide ingestion of pre-computed embeddings
+        if self.overwrite:
+            self.client.schema.delete_class(self.class_name)
+            self.client.schema.create_class(self.class_obj)
+        elif not self.client.schema.exists(self.class_name):
+            self.client.schema.create_class(self.class_obj)
+        else:
+            # do not index
+            return dataframe
+
+        self.client.batch.configure(batch_size=100, dynamic=True, num_workers=2)
```

Comment:
Would be nice if we can also include some of those parameters in the spec as opposed to hardcoding them. You can set the current values as default